### ISSUE TITLE
RequireTernaryOperatorSniff false positive on reference

### ISSUE THREAD

**Issue by sebastienbarre** [slevomat/coding-standard#826]
The following code triggers " Use ternary operator. "

```php
    $condition = true;
    $foo = 1;
    $bar = 2;

    if ($condition) {
      $ref = &$foo;
    } else {
      $ref = &$bar;
    }
```

However, it is illegal to use a ternary operator with refs in PHP:
```
    $ref = $condition ? &$foo : &$bar;
```
will trigger "unexpected '&' in Standard input code"

**Comment by sebastienbarre** [comment]
@grongor @kukulich Thanks for the quick fix.

Now I wanted to mention something that has been pointed out to me since then.

This is illegal in PHP:
```php
$ref = $condition ? &$foo : &$bar;
```
but this is legal apparently:
```php
$ref = &${$condition ? 'foo' : 'bar'};
```

Your call. Personally I find it pretty ugly, so I'm not sure I would want to force people to do it.

**Comment by grongor** [comment]
oh, man ... yeah, I wouldn't mention it anywhere and just hope that everybody else will eventually forget it's possible :joy:
